src/main/java/ml/shifu/shifu/core/dtrain/dt/DTMaster.java

```diff
@@ -328,6 +332,13 @@ public DTMasterParams doCompute(MasterContext<DTMasterParams, DTWorkerParams> co
         Map<Integer, TreeNode> todoNodes = new HashMap<Integer, TreeNode>();
         DTMasterParams masterParams = new DTMasterParams(weightedTrainCount, trainError, weightedValidationCount,
                 validationError);
+
+        // Add  early Stop Feature for GBT
+        if(this.earlyStopEnabled && this.isGBDT && dtEarlyStopDecider.add(trainError, validationError)){
+            masterParams.setHalt(true);
+            LOG.info("Early stop identified, training is stopped in iteration {}.", context.getCurrentIteration());
```

Comment:
I have also considered to return here. But finally not return for not familiar with the code. I am afraid if I return here, then these code such as "masterParams.setTrees(trees);", "doCheckPoint(context, masterParams);" will not be executed, will this caused trouble? After all, the following code has no where setHalt to false.
It better return if no trouble will cause. I will check this with you tomorrow.